crates/moon/src/cli/run.rs

```diff
@@ -388,13 +388,14 @@ pub fn run_run_internal(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Res
         dynamic_stub_libs: None,
     };
 
-    let mut module = scan_with_x_build(
+    let mut module = scan_with_x_build_single_package(
         false,
         &moonc_opt,
         &moonbuild_opt,
         &resolved_env,
         &dir_sync_result,
         &PrePostBuild::PreBuild,
+        &package,
     )?;
 
     let pkg = module.get_package_by_path_mut(&package).unwrap();
```

Comment:
IMO setting `filter_package` in `moonbuild_opt.BuildOpt` is enough
```rust
    moonbuild_opt.build_opt = Some(moonutil::common::BuildOpt {
        filter_package: Some(pkg.full_name()),
        ..Default::default()
    });
```

Reply:
Thanks a lot! It does the job greatly.